File: .NET/TesteWebApi/TesteWebApi/models/Produto.cs
using System.Text.Json.Serialization;

namespace TesteWebApi.models
{
    public class Produto
    {
        public int Id { get; set; }
        public string Nome { get; set; } = "";
        public string Descricao { get; set; } = "";

        public int CategoriaId { get; set; }

        [JsonIgnore]
        public Categoria? Categoria { get; set; }
    }
}

File: NodeJs/ApiNode/src/models/produto.js
const Sequelize = require('sequelize');
const database = require('../database/db');
const Categoria = require('../models/categoria')

const Produto = database.define('produto',{
    nome:{
        type: Sequelize.STRING,
        allowNull:false
    },
    preco:{
        type: Sequelize.DOUBLE
    },
    descricao:{
        type: Sequelize.STRING
    }
});

Produto.belongsTo(Categoria,{
    foreignKey: 'categoriaId',
    as: 'categoria'
});

Categoria.hasMany(Produto, { 
    foreignKey: 'categoriaId',
    as: 'produtos'
});

module.exports = Produto;